dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/security/Authorizer.java

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal.security;
+
+import io.openliberty.mcp.internal.McpTransport;
+import io.openliberty.mcp.internal.ToolMetadata;
+import io.openliberty.mcp.internal.exceptions.jsonrpc.HttpResponseException;
+import jakarta.servlet.http.HttpServletResponse;
+
+/**
+ * Used for user authorization for calling and listing of tools
+ */
+public class Authorizer {
+
+    private static final boolean AUTHORIZED = true;
+    private static final boolean NOT_AUTHORIZED = false;
+
+    /**
+     *
+     * Authorizes the user without throwing an exception if authorization fails
+     *
+     * @param transport
+     * @param tmd
+     * @return true if the user has been authorized
+     */
+    public static boolean authorized(McpTransport transport, ToolMetadata tmd) {
+        return doAuthorization(transport, tmd);
+    }
+
+    /**
+     *
+     * Authorizes the user and throws an exception if the user is not authorized
+     *
+     * @param transport
+     * @param tmd
+     * @return true if the user has been authorized
+     */
+    public static boolean requireAuthorized(McpTransport transport, ToolMetadata tmd) {
+        if (doAuthorization(transport, tmd)) {
+            return true;
+        } else {
+            throw new HttpResponseException(HttpServletResponse.SC_FORBIDDEN);
+        }
+    }
```

Comment:
Since this method throws an exception if the condition it checks is not met, it should not return a `boolean`.

We can also improve the Javadoc

```suggestion
    /**
     * Throws an exception if the user is not authorized to access a particular MCP tool
     *
     * @param transport the McpTransport
     * @param tmd the tool to check access for
     * @throws HttpResponseException if the user is not authorized
     */
    public static void requireAuthorized(McpTransport transport, ToolMetadata tmd) {
        if (!doAuthorization(transport, tmd)) {
            throw new HttpResponseException(HttpServletResponse.SC_FORBIDDEN);
        }
    }
```

Reply:
Javadoc here is not correct